internal/api/mfa.go

```diff
@@ -626,7 +639,7 @@ func (a *API) VerifyFactor(w http.ResponseWriter, r *http.Request) error {
 				return terr
 			}
 		}
-		if shouldReEncrypt && config.Security.DBEncryption.Encrypt && factor.IsTOTPFactor() {
+		if shouldReEncrypt && config.Security.DBEncryption.Encrypt {
```

Comment:
Not needed since only TOTP factors should reach this code path.

For avoidance of doubt, we may want to consider putting all TOTP related logic into a function similar to what we are doing with `WebAuthn` and `Phone`